### PR TITLE
Write a jet for +sort

### LINKED ISSUE #387: ++sort jet not implemented
This jet is just a stub right now. Not a totally trivial jet to write, but probably a good introductory task for someone who's interested in jetting.

### PR DESCRIPTION
**Pull request by eglaysher**
The majority of time in `+gather-builds` in Ford was spent inside `+sort`.

Closes #387

### CODE REVIEW

File: jets/b/sort.c [hunk 1]
```diff
@@ -6,13 +6,70 @@
...
+      u3_noun hoz = u3j_gate_slam(sit_u, u3nc(u3k(u3h(a)), u3k(second)));
+      u3_noun nex;
+
+      if ( c3y == hoz ) {
+        nex = u3nc(u3nc(u3h(a), u3k(u3h(acc))), u3k(u3t(acc)));
+      }
+      else {
+        nex = u3nc(u3k(u3h(acc)), u3nc(u3h(a), u3k(u3t(acc))));
+      }
+      u3z(hoz);
```
Comment (belisarius222) [inline jets/b/sort.c]: I think we need to also `u3z(a)` here.

File: jets/b/sort.c [hunk 2]
```diff
@@ -6,13 +6,70 @@
...
+            u3_noun a,
+            u3_noun second)
+  {
+    if ( 0 == a ) {
+      return u3nc(u3_nul, u3_nul);
+    }
+    else if ( c3n == u3du(a) ) {
+      return u3m_bail(c3__exit);
+    } else {
+      u3_noun acc = _split_in(sit_u, u3t(a), second);
```
Comment (belisarius222) [inline jets/b/sort.c]:
```
u3_noun hed = u3k(u3h(a));
u3_noun tal = u3k(u3t(a));
u3z(a);
```

Then use `hed` and `tal` in the rest of the function.

**Comment by ohAitch** [comment]
Alternately, u3x_cell(a, *hed, *tal)

**Comment by joemfb** [comment]
It worked! How much faster are things with `+sort` jetted?

File: jets/b/sort.c [hunk 3]
```diff
@@ -6,13 +6,73 @@
 
 /* functions
 */
+  // like skid, except its callback is $-([* *] ?) and it takes the second
+  // argument so that it calls its callback with [i.list, second]
+  //
+  // second is RETAINED
```
Comment (belisarius222) [inline jets/b/sort.c]: All the args are retained

**Comment by eglaysher** [comment]
Things are faster in +gather-builds. For the degenerate case I was measuring before (first uncached build of a webpage), it's gone from 200ms to 36ms.

(I assume everything else that uses sort is also faster, but I haven't measured.)

**Comment by eglaysher** [comment]
This is now ready for real review. PTAL.

**Pull request merged**